### PR TITLE
WIP: shim design example

### PR DESCRIPTION
**Pull request by andreamah**
DON'T MERGE; just an small-scale example of the python microbit model design (as requested by coaches)

### CODE REVIEW

**Comment by andreamah** [comment]
@sagarmanchanda is this what you envisioned in terms of the shim-based singleton design?

**Comment by sagarmanchanda** [comment]
> @sagarmanchanda is this what you envisioned in terms of the shim-based singleton design?

**Review by sagarmanchanda — COMMENTED** [review]
Looks good. This is exactly what I was thinking. 

Although we should now think about the folder structuring too.

src
|____microbit
_____|____tests
_____|____microbit_model
__________|____microbit_model.py
__________|____display # Could be a folder if it has multiple files, otherwise just files in root
_____|____shim.py
_____|____init.py
       
What do you guys think about the folder structuring?

**Comment by andreamah** [comment]
@sagarmanchanda I think that that directory structure sounds good! I'll try to incorporate that into the upcoming work.

Also, I was thinking that it might be better to replace 

```
class display:
    @staticmethod
    def scroll(self, message):
        microbit_model.mb.display.scroll(self,message) 
```
in src/microbit/code_processing_shim.py with 

`display = microbit_model.mb.display`

To reduce repetition with static methods mapping to actual object methods. I believe that the new version uses pass-by-reference in order to assign a simple global "display" object with the microbit model's display.

What do you think?

File: src/microbit/code_processing_shim.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from . import microbit_model
+
+# EXAMPLE
+# can be called simply as "show_message("string")"
+def show_message(message):
```
Comment (xnkevinnguyen) [inline src/microbit/code_processing_shim.py:7]: That's awesome, it was exactly what I was thinking as well :)

**Comment by sagarmanchanda** [comment]
@andreamah Good catch! Totally makes sense to directly use the methods rather than creating static mappings.

**Comment by sagarmanchanda** [comment]
Also thanks for doing this sample implementation! Good job. :)

**Comment by andreamah** [comment]
no intention to merge these changes yet, will ping coaches with new changes when it comes up

**Pull request closed without merging**